spk/firebird/src/service-setup.sh

```diff
@@ -0,0 +1,105 @@
+# Fbguard binary is a wrapper around firebird that allows it to run as
+# a daemon
+FBGUARD_BIN_FILE="${SYNOPKG_PKGDEST}/bin/fbguard"
+FB_SECDB="${SYNOPKG_PKGVAR}/security4.fdb"
+
+# We set onetime because we want the systemd to take care about the failures
+SERVICE_COMMAND="${FBGUARD_BIN_FILE} -pidfile ${PID_FILE} -daemon -onetime"
+
+SYSDBA_PASSWORD_FILE="${SYNOPKG_PKGVAR}/SYSDBA.password"
+
+validate_preupgrade () {
+    # We must check that either SYSDBA.password or the security database
+    # exists.
+    # If neither of them is present, we don't know how to set the main
+    # DB password.
+    [ -f "$SYSDBA_PASSWORD_FILE" ] || [ -f "$FB_SECDB" ] || {
+        install_log "Neither SYSDBA.password nor security4.fdb exists.";
```

Comment:
you must not use `install_log` but `echo` to force the display of the validation error message.
```suggestion
        echo "Neither SYSDBA.password nor security4.fdb exists."
```